@here/olp-sdk-dataservice-api/lib/blob.v2.api.ts

```diff
@@ -0,0 +1,613 @@
+ * you can enable the optional capability of listing which keys are stored in the layer. This capability has a negative impact
+ * on the access times when getting and saving blobs and therefore must be enabled explicitly.
+ *
+ * OpenAPI spec version: 2.0.0
+ *
+ *
+ * NOTE: This class is auto generated by the swagger code generator program.
+ * https://github.com/swagger-api/swagger-codegen.git
+ *
+ * Do not edit the class manually.
+ */
+
+import { UrlBuilder, RequestBuilder, RequestOptions } from "./RequestBuilder";
+
+/**
+ * The identifier of the blob.
+ */
+export interface DataHandleResponse {
+    /**
+     * The identifier of the blob.
+     */
+    handle: string;
+}
+
+/**
+ * Object or common prefix stored in Object Store layer.
+ */
+export interface KeysListObjectItemResponse {
+    /**
+     * Name of the object/common prefix.
+     */
+    name: string;
+    /**
+     * Object size in bytes or omitted if the `type` field is `commonPrefix`.
+     */
+    size?: number;
+    /**
+     * Last modified date and time in RFC 3339 format or omitted if the `type` field is `commonPrefix`.
+     */
+    lastModified?: Date;
+    /**
+     * Indicates whether the item is object or common prefix.
+     */
+    type: TypeEnum;
+}
+
+export type TypeEnum = "object" | "commonPrefix";
+/**
+ * Paginated response of keys and common prefixes.
+ */
+export interface KeysListResponse {
+    /**
+     * Page token to fetch the next page.
+     */
+    pageToken: string;
+    items: KeysListObjectItemResponse[];
+}
+
+export interface MultipartCompletePart {
+    /**
+     * The identifier assigned by the system to identify the uploaded part.
+     * The `id` is returned by the upload part operation after successfully uploading a part.
+     */
+    id: string;
+    /**
+     * The number of the part of the multipart upload.
+     * This is the same number used in the `partNumber` parameter in the upload part operation.
+     * If an error occurs during upload, do not reuse the `partNumber` when retrying the upload.
+     */
+    number: number;
+}
+
+export interface MultipartCompleteRequest {
+    parts?: MultipartCompletePart[];
+}
+
+/**
+ * Describes a status of the multipart upload.
+ */
+export interface MultipartHandleUploadStatus {
+    /**
+     * The status of the multipart upload.
+     */
+    status: StatusEnum;
+    /**
+     * The data handle that identifies a specific blob so that you can get that blob's contents.
+     * Present only when the upload is complete.
+     */
+    handle?: string;
+}
+
+/**
+ * The identifier of the multipart upload (token).
+ */
+export interface MultipartInitResponse {
+    /**
+     * The identifier of the multipart upload (token).
+     */
+    multipartToken: string;
+}
+
+/**
+ * Describes a status of the multipart upload.
+ */
+export interface MultipartKeyUploadStatus {
+    /**
+     * The status of the multipart upload.
+     */
+    status: StatusEnum;
```

Comment:
Same here. Not sure if we should do it like this. The risk is that we need to update in case one value is added. And this might happen without notice and without REST API version change. I would instead propose to add just string.

Reply:
Changed to string, thx